e2etest/l2tests/interface_selector.go

```diff
@@ -173,5 +175,213 @@ var _ = ginkgo.Describe("L2-interface selector", func() {
+
+			for i := range NodeNics {
+				l2Adv := metallbv1beta1.L2Advertisement{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("with-interfaces-%d-l2adv", i),
+					},
+					Spec: metallbv1beta1.L2AdvertisementSpec{
+						IPAddressPools: []string{"l2-test"},
+						Interfaces:     []string{NodeNics[0]},
+					},
+				}
+				resources.L2Advs = append(resources.L2Advs, l2Adv)
+			}
+
+			err := mac.FlushIPNeigh(ingressIP, executor.Host)
+			framework.ExpectNoError(err)
+			err = ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			gomega.Eventually(func() error {
+				return mac.RequestAddressResolutionFromIface(ingressIP, LocalNics[0], executor.Host)
+			}, 10*time.Second, 1*time.Second).ShouldNot(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("node selector", func() {
+			svc, _ := service.CreateWithBackend(cs, f.Namespace.Name, "lb-service")
+			defer func() {
+				err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+				framework.ExpectNoError(err)
+			}()
+
+			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+
+			framework.ExpectNoError(err)
+			for _, node := range allNodes.Items {
+				resources := internalconfig.ClusterResources{
+					L2Advs: []metallbv1beta1.L2Advertisement{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "with-interfaces",
+							},
+							Spec: metallbv1beta1.L2AdvertisementSpec{
+								Interfaces:    []string{NodeNics[0]},
+								NodeSelectors: k8s.SelectorsForNodes([]v1.Node{node}),
+							},
+						},
+					},
+				}
+
+				err = ConfigUpdater.Update(resources)
+				framework.ExpectNoError(err)
+
+				ingressIP := e2eservice.GetIngressPoint(&svc.Status.LoadBalancer.Ingress[0])
+				err := mac.FlushIPNeigh(ingressIP, executor.Host)
+				framework.ExpectNoError(err)
+
+				var svcNode *corev1.Node
+				gomega.Eventually(func() error {
+					svcNode, err = k8s.GetSvcNode(cs, svc.Namespace, svc.Name, allNodes)
+					return err
+				}, 1*time.Minute, 1*time.Second).Should(gomega.Not(gomega.HaveOccurred()))
+				speakerPod, err := metallb.SpeakerPodInNode(cs, svcNode.Name)
+				framework.ExpectNoError(err)
+				selectorMac, err := mac.GetIfaceMac(NodeNics[0], executor.ForPod(speakerPod.Namespace, speakerPod.Name, "speaker"))
+				framework.ExpectNoError(err)
+				if node.Name == speakerPod.Name {
+					gomega.Eventually(func() string {
+						err := mac.RequestAddressResolutionFromIface(ingressIP, LocalNics[0], executor.Host)
+						if err != nil {
+							return err.Error()
+						}
+						err = service.ValidateL2(svc)
+						if err != nil {
+							return err.Error()
+						}
+						ingressMac, err := mac.ForIP(ingressIP, executor.Host)
+						if err != nil {
+							return err.Error()
+						}
+						return ingressMac.String()
+
+					}, 1*time.Minute, 1*time.Second).Should(gomega.Equal(selectorMac.String()))
+				}
+			}
+		})
+
+		ginkgo.It("Modify Address pool", func() {
```

Comment:
I don't think this test is useful. We create one service in the beginning, it gets the ip assigned, we add a new ip to the pool but the service doesn't change